Journalist can create article

erikbjoern opened the pull request:
# User story 
[PT Link](https://www.pivotaltracker.com/story/show/172888071)
```
As a journalist,
In order to write articles
I would like to have an article creation form  
```
# Feature
* Journalist can create articles
* With title and body, both fields required
* Displays error message from API when any field is left out
* They are posted to the main news site as they are submitted 

# Additions in this PR
* `CreateArticle` component (the page where journalist write an article)
* `WritingForm` component (placed on the `CreateArticle` page)
* `Header` component (always visible on top of site)
* Scripts for running Cypress tests properly

## Packages added
* axios
* semantic-ui-css
* semantic-ui-react
* start-server-and-test

# Screenshots
![Screenshot 2020-05-24 at 11 05 24](https://user-images.githubusercontent.com/61984272/82750167-8c2f4580-9dae-11ea-9011-6d432647c4c4.png)

erikbjoern committed:
writing feature test for creating article, including fixture from api documentation

PaulineBA committed:
CreateArticle component

PaulineBA committed:
adding form to component

erikbjoern committed:
modifying test and the elements tested, adding stylesheet from user client

erikbjoern committed:
adding axios and post request on submit, successfully hitting fixture when testing

kermit-klein committed:
add message field for response,add start-server-and-test package

erikbjoern committed:
applying basic styling

oliverochman commented on src/components/CreateArticle.jsx:
```diff
@@ -0,0 +1,39 @@
+  return (
+    <Container className="writing-container">
+      <Form onSubmit={onSubmitHandler}>
+        <Form.Field>
```
I need u to extract the form to a separate component

oliverochman commented on cypress/integration/journalistCanCreateArticle.feature.js:
```diff
@@ -0,0 +1,34 @@
+      "This is the body this is the body this is the body this is the body this is the body."
+    );
+    cy.get("#post").click();
+    cy.get("#message").should("not.be.visible");
```
You need to test for a error message here.

erikbjoern replied:
The input fields have the 'required' property, so we can't send off an incomplete request. Should we change that?

oliverochman commented on cypress/integration/journalistCanCreateArticle.feature.js:
```diff
@@ -0,0 +1,34 @@
+  it("unsuccessfully without entering any body text", () => {
+    cy.get("input#title").type("This is the title");
+    cy.get("#post").click();
+    cy.get("#message").should("not.be.visible");
```
same here

erikbjoern committed:
resolving PR comments
removing 'required' and instead showing error message from back-end
extracting form to separate component
updating Cypress version to try get Semaphore testing to work

oliverochman commented on src/components/CreateArticle.jsx:
```diff
@@ -0,0 +1,14 @@
+import { Container } from 'semantic-ui-react'
+
+const CreateArticle = () => {
+
```
this is not what I meant by extracting it to another component.

oliverochman replied:
please only extract the form

erikbjoern committed:
moving onSubmitHandler back to CreateArticle component

ViaMarcus commented on src/components/CreateArticle.jsx:
```diff
@@ -0,0 +1,27 @@
+  const onSubmitHandler = async (e) => {
+    try {
+      const response = await axios.post("/articles", {
+        params: { title: e.target.title.value, body: e.target.body.value },
```
Unwrap the title and body, params {} wrapper does not work with backend.

erikbjoern committed:
unwrapping article params

kermit-klein commented on src/components/CreateArticle.jsx:
```diff
@@ -9,7 +9,7 @@ const CreateArticle = () => {
     try {
       const response = await axios.post("/articles", {
-        params: { title: e.target.title.value, body: e.target.body.value },
+        title: e.target.title.value, body: e.target.body.value
```
makes sense

designerofthing reviewed (approved):
looks good
